editor/gui/scene_tree_editor.cpp

```diff
@@ -1531,6 +1548,23 @@ void SceneTreeDialog::set_valid_types(const Vector<StringName> &p_valid) {
 		label->set_text(type);
 		label->set_auto_translate(false);
 	}
+
+	// Extract existing filter from VBoxContainer and add to HBoxContainer.
+	HBoxContainer *hbc = memnew(HBoxContainer);
+	Node *filter_container = filter->get_parent();
+	filter_container->remove_child(filter);
+	hbc->add_child(filter);
+
+	// Set HBoxContainer as 2nd child of VBoxContainer.
+	filter_container->add_child(hbc);
+	filter_container->move_child(hbc, 1);
+
+	// Add 'Show All' button to HBoxContainer next to the filter.
+	CheckButton *show_all_nodes = memnew(CheckButton);
+	show_all_nodes->set_text(TTR("Show All"));
+	show_all_nodes->connect("toggled", callable_mp(this, &SceneTreeDialog::_show_all_nodes_changed));
+	show_all_nodes->set_h_size_flags(Control::SIZE_SHRINK_BEGIN);
+	hbc->add_child(show_all_nodes);
```

Comment:
This solution is quite hacky. You could add the HBoxContainer in the constructor and only add the CheckButton here.
(or you could even add the CheckButton in the constructor and hide it until needed)

Reply:
CheckButton is now always created in the constructor, but only visible when valid_types is defined.